nevergrad/benchmark/experiments.py

```diff
@@ -40,25 +40,19 @@ def mltuning(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
     # Continuous case
 
     # First, a few functions with constraints.
-    functions = [  # Tool and dimension.
-        MLTuning("decision_tree", 3),
-        MLTuning("decision_tree_depth", 3),
-        MLTuning("mlp", 2),
-        MLTuning("mlp", 1),
-        MLTuning("mlp", 3),
-        MLTuning("any", 3),
-        MLTuning("any", 1),
-    ]
     optims = ["Shiwa", "DE", "DiscreteOnePlusOne", "PortfolioDiscreteOnePlusOne", "CMA", "MetaRecentering",
               "DoubleFastGADiscreteOnePlusOne"]
     for budget in [50, 150, 500]:
         for num_workers in [1, 10, 50, 100]:
             for optim in optims:
-                for function in functions:
-                    xp = Experiment(function, optim, num_workers=num_workers,
-                                    budget=budget, seed=next(seedg))
-                    if not xp.is_incoherent:
-                         yield xp
+                for dimension in [1, 2, 3]:
+                    for regressor in ["mlp", "decision_tree", "decision_tree_depth"]:
+                        function = MLTuning(regressor, dimension)
+                        function.add_descriptors(regressor=regressor, dimension=dimension)
```

Comment:
the descriptors should be directly implemented *inside* the function

Reply:
Done.